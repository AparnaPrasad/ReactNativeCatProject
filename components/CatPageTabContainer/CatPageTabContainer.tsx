import React from 'react';
import { StyleSheet, View, ScrollView } from 'react-native';
import { TabIdEnum } from '../../utils/constants';
import PageTabs from '../PageTabs/PageTabs';
import TabContentContainer from '../TabContentContainer/TabContentContainer';
import useWidthHeight from '../ScreenWidthHeightProvider/ScreenWidthHeightProvider';
import { ApplicationState } from '../../store';
import { useSelector } from 'react-redux';

interface ReduxProps {
    backgroundColor: string
}

export default function CatPageTabContainer() {
    const [selectedTab, setSelectedTab] = React.useState(0)
    const scrollRef = React.useRef(null);
    const { screenHeight } = useWidthHeight();
    //const effectiveHeight = screenHeight;
    const { backgroundColor } = useSelector<ApplicationState, ReduxProps>((state: ApplicationState) => {
        return {
            backgroundColor: state.pageState.backgroundColor,
        }
    });

    const selectTab = (tabId: TabIdEnum) => {
        setSelectedTab(0);
        if (scrollRef && scrollRef.current) {
            scrollRef.current.scrollTo({ x: 0, y: (tabId * screenHeight), animated: true })
        }
    }

    return (
        <View style={[styles.container, { backgroundColor }]}>
            <PageTabs selectedTab={selectedTab} selectTab={selectTab} />
            <ScrollView contentContainerStyle={styles.scrollViewContainerStyle} style={styles.scrollView} ref={scrollRef} >
                <TabContentContainer/>
            </ScrollView>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
    },
    scrollView: {
        width: '100%'
    },
    scrollViewContainerStyle: {
        flexGrow: 1,
        justifyContent: 'space-between'
    }
});