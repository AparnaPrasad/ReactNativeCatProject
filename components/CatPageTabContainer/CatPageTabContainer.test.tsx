import { shallow } from "enzyme";
import React from 'react';
import CatPageTabContainer from './CatPageTabContainer';
import { View } from "react-native";
import { Provider } from "react-redux";
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

const mockStore = configureStore([]);

describe('it should render app without crashing', () => {

    it('should render a view with a background color', () => {
        const wrapper = shallow(<View style={{ backgroundColor: '#fff' }} />)
        expect(wrapper).toMatchSnapshot()
    })

    it('should render a view with background color in the redux store', () => {
        const backgroundColor ="#fff"
        let store = mockStore({
            pageState: {
                backgroundColor: backgroundColor
            }
        });

        const CatPageTabContainerRenderer = renderer.create(<Provider store={store}> <CatPageTabContainer /></Provider >)
        const CatPageTabContainerRoot = CatPageTabContainerRenderer.root;
        expect(CatPageTabContainerRoot.findByType(View).props.style).toEqual(expect.arrayContaining([{ backgroundColor}]));
    })
})

