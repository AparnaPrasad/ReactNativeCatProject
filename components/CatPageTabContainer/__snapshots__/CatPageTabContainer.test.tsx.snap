// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render app without crashing should render a view with a background color 1`] = `ShallowWrapper {}`;
